### ISSUE TITLE
UI Widget Functionality Missing

### ISSUE THREAD

**Issue by dewang-mistry** [vercel/ai-chatbot#437]
The latest code update has removed all custom UI widgets that were displayed when requesting stock prices. It appears that all related functionality has been eliminated. Was this removal intentional, or was it a mistake?

**Comment by athrael-soju** [comment]
Looks intentional to me and for good reason. This is now a proper, clean template, ready for Next 15 + react 19, among several other goodies.

**Comment by dewang-mistry** [comment]
I apologize for my basic questions. I'm new to React and this framework because I focus on data science and machine learning. I'm working on a project where I want to showcase the output of my data in a chat interface using widgets. 

Is it possible for me to build the widgets and integrate them into my project? Thank you for your support and for answering my questions.

**Comment by jeremyphilemon** [comment]
@dewang-mistry you're right that the previous stock components are missing – we just updated the template, you can read about it [here](https://github.com/vercel/ai-chatbot/issues/438).

The updated template also supports adding widgets, you can try asking the weather for it to show up! Here's the `getWeather` tool that you can then use to  render a `<Weather/>` component in the chat interface. For your use case, you will have to do the same thing by adding a tool + component to render the output of your data!
